server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -193,52 +184,7 @@ public void registerRepository(final PutRepositoryRequest request, final ActionL
-            private boolean changed = false;
-
-            @Override
-            public ClusterState execute(ClusterState currentState) {
-                Metadata metadata = currentState.metadata();
-                Metadata.Builder mdBuilder = Metadata.builder(currentState.metadata());
-                RepositoriesMetadata repositories = metadata.custom(RepositoriesMetadata.TYPE, RepositoriesMetadata.EMPTY);
-                List<RepositoryMetadata> repositoriesMetadata = new ArrayList<>(repositories.repositories().size() + 1);
-                for (RepositoryMetadata repositoryMetadata : repositories.repositories()) {
-                    if (repositoryMetadata.name().equals(newRepositoryMetadata.name())) {
-                        Repository existing = RepositoriesService.this.repositories.get(request.name());
-                        if (existing == null) {
-                            existing = RepositoriesService.this.internalRepositories.get(request.name());
-                        }
-                        assert existing != null : "repository [" + newRepositoryMetadata.name() + "] must exist";
-                        assert existing.getMetadata() == repositoryMetadata;
-                        final RepositoryMetadata updatedMetadata;
-                        if (canUpdateInPlace(newRepositoryMetadata, existing)) {
-                            if (repositoryMetadata.settings().equals(newRepositoryMetadata.settings())) {
-                                // Previous version is the same as this one no update is needed.
-                                return currentState;
-                            }
-                            // we're updating in place so the updated metadata must point at the same uuid and generations
-                            updatedMetadata = repositoryMetadata.withSettings(newRepositoryMetadata.settings());
-                        } else {
-                            ensureRepositoryNotInUse(currentState, request.name());
-                            updatedMetadata = newRepositoryMetadata;
-                        }
-                        found = true;
-                        repositoriesMetadata.add(updatedMetadata);
-                    } else {
-                        repositoriesMetadata.add(repositoryMetadata);
-                    }
-                }
-                if (found == false) {
-                    repositoriesMetadata.add(new RepositoryMetadata(request.name(), request.type(), request.settings()));
-                }
-                repositories = new RepositoriesMetadata(repositoriesMetadata);
-                mdBuilder.putCustom(RepositoriesMetadata.TYPE, repositories);
-                changed = true;
-                return ClusterState.builder(currentState).metadata(mdBuilder).build();
-            }
-
+        submitUnbatchedTask("put_repository [" + request.name() + "]", new RegisterRepositoryTask(this, request, acknowledgementStep) {
```

Comment:
This refactoring simply moves the code so we can reuse the task execute method for the reserved state handler.